include/wx/event.h

```diff
@@ -4065,8 +4067,18 @@ class WXDLLIMPEXP_BASE wxEvtHandler : public wxObject
     wxEvtHandler*       m_nextHandler;
     wxEvtHandler*       m_previousHandler;
 
-    typedef wxVector<wxDynamicEventTableEntry*> DynamicEvents;
-    DynamicEvents* m_dynamicEvents;
+    struct DynamicEvents
+    {
+        wxVector<wxDynamicEventTableEntry*> m_entries;
+        wxRecursionGuardFlag m_flag = 0;
+    };
+    // use wxSharedPtr so that SearchDynamicEventTable() can use another
+    // instance of wxSharedPtr to extend the life of the wxRecursionGuardFlag
+    // to outlive wxRecursionGuard
+    wxSharedPtr<DynamicEvents> m_dynamicEvents;
+    // ensure new m_dynamicEvents has same layout as wxWidgets 3.2 m_dynamicEvents
+    static_assert(sizeof(wxSharedPtr<DynamicEvents>) == sizeof(DynamicEvents*), "wxSharedPtr<> has wrong size");
+    static_assert(alignof(wxSharedPtr<DynamicEvents>) == alignof(DynamicEvents*), "wxSharedPtr<> has wrong alignment");
```

Comment:
Just FYI, I'll remove this because we can't backport this to 3.2 anyhow (as we can't even use C++11 there) and this is not ABI compatible in any case because `DynamicEvents` has changed.

Reply:
I don't need 3.2 support, but I think it would be nice to fix this problem there as well.  Are you sure this isn't ABI compatible?  `DynamicEvents` is not referenced using inline functions, so I think its layout should only matter to event.cpp, not an application that's dynamic linked to wxWidgets.  The issue of `static_assert` being unavailable in 3.2 is unfortunate, but we could at least put `wxASSERT()`s somewhere in event.cpp for (only) 3.2.

Reply:
I'm not sure that this is ABI-incompatible, but I suspect it is and I'm almost sure this is going to provoke errors from the ABI checker tool that we use in 3.2 (you can try making a PR with just this change against 3.2 to run it).

Reply:
I have created https://github.com/wxWidgets/wxWidgets/pull/25378 which successfully backports this PR to 3.2.